### PR TITLE
[tests-only][full-ci] Add CORS API tests

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Skipping core CORS API tests in https://github.com/owncloud/core/pull/40527
So adding cors tests locally here.

## Related Issue
- part of https://github.com/owncloud/core/issues/40104
- Depends on: https://github.com/owncloud/core/pull/40527

## How Has This Been Tested?
- test environment:

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by saw-jan** [comment]
Welcoming reviews & discussion

**Comment by individual-it** [comment]
CC @ScharfViktor what do you think?

**Comment by ScharfViktor** [comment]
> CC @ScharfViktor what do you think?

we use extra env "OCIS_CORS_ALLOW_ORIGINS": "https://aphno.badal" in the `apiCors` 
- if we run this test localy we should start ocis with `OCIS_CORS_ALLOW_ORIGINS` env. Should we mention it at the beginning of the test?
- if we want to move new tests from core to `apiCors` It's not like we can change Env?

File: tests/acceptance/features/apiCors/cors.feature [hunk 1]
```diff
@@ -0,0 +1,66 @@
+@api @skipOnOcV10
+Feature: CORS headers
```
Comment (ScharfViktor) [inline tests/acceptance/features/apiCors/cors.feature:2]: add please description: if you want to run test locally - you should run ocis with env "OCIS_CORS_ALLOW_ORIGINS": "https://aphno.badal"

**Review by ScharfViktor — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiCors/cors.feature [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      | Access-Control-Allow-Methods  |
+    Examples:
+      | ocs_api_version | endpoint                          | ocs-code | http-code |
+      | 1               | /config                           | 100      | 200       |
+      | 2               | /config                           | 200      | 200       |
+      | 1               | /apps/files_sharing/api/v1/shares | 100      | 200       |
+      | 2               | /apps/files_sharing/api/v1/shares | 200      | 200       |
+
+
+  Scenario Outline: CORS headers should be returned when invalid password is used
```
Comment (SwikritiT) [inline tests/acceptance/features/apiCors/cors.feature:54]:
```suggestion
  Scenario Outline: CORS headers should be returned when an invalid password is used
```

File: tests/acceptance/features/apiCors/cors.feature [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#
+# NOTE: for running this feature locally, you need to run oCIS with the following env var:
+# CORS_ALLOWED_ORIGINS=https://aphno.badal
+#
```
Comment (SwikritiT) [inline tests/acceptance/features/apiCors/cors.feature:4]:
```suggestion
# NOTE: for running this feature locally, you need to run oCIS with the following env var:
# CORS_ALLOWED_ORIGINS=https://aphno.badal
```
Why these extra `#` ?

Comment (saw-jan) [reply tests/acceptance/features/apiCors/cors.feature:4]: just like you notice it, I want others to notice the odd and read the NOTE :smile:

File: tests/acceptance/features/apiCors/cors.feature [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      | Access-Control-Allow-Methods  | GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
+    Examples:
+      | ocs_api_version | endpoint                          | ocs-code | http-code |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiCors/cors.feature:66]: ocs-code is same for both examples

**Review by SwikritiT — COMMENTED** [review]
lgtm

File: tests/acceptance/features/apiCors/cors.feature [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      | 1               | /config                           | 100      | 200       |
+      | 2               | /config                           | 200      | 200       |
+      | 1               | /apps/files_sharing/api/v1/shares | 100      | 200       |
+      | 2               | /apps/files_sharing/api/v1/shares | 200      | 200       |
+
+
+  Scenario Outline: no CORS headers should be returned when CORS domain does not match Origin header
```
Comment (grgprarup) [inline tests/acceptance/features/apiCors/cors.feature:31]:
```suggestion
  Scenario Outline: CORS headers should not be returned when CORS domain does not match Origin header
```

**Review by individual-it — APPROVED** [review]

**Pull request merged**